### DISCUSSION TITLE
Input type not supported error when adding boolean values

### DISCUSSION THREAD

**Discussion by kinekt4** [downforcetech/fluidxml.php#49] [Questions]
When calling `addChild()`, if any values of the array contains a boolean, then FluidXML will return a " Input type not supported" error.

Error:

    $xml = new FluidXml();
    $xml->addChild(array('bool' => true));
    
No Error:

    $xml = new FluidXml();
    $xml->addChild(array('bool' => 'true'));

I don't know what the solution should be.

Perhaps it should be casted to the string value 'true' and 'false'.

**Comment by daniele-orlando** [comment]
Or it should be casted to the string `'yes'` or to the string `'ok'` or to the integer `1`...

A Boolean value is not a valid Node Name nor a valid Node Value, the only always correct way to handle this opinionated case is raising an error and leave the choice on how to handle the conversion to the user.

**Comment by kinekt4** [comment]
Thanks for your response. It has been useful.
